frontend/src/composables/utils.ts

```diff
@@ -0,0 +1,294 @@
+          type: "NULL",
+        };
+      }
+      if (Array.isArray(value)) {
+        return {
+          value: RowValue.fromPartial({
+            stringValue: JSON.stringify(value.map(flattenNoSQLColumn)),
+          }),
+          type: "OBJECT",
+        };
+      }
+      if (value instanceof Date) {
+        return {
+          value: RowValue.fromPartial({
+            timestampValue: Timestamp.fromPartial({
+              seconds: Math.floor(value.getTime() / 1000),
+              nanos: (value.getTime() % 1000) * 1e6,
+            }),
+          }),
+          type: "DATETIME",
+        };
+      }
+
+      if (nested) {
+        const formatted = flattenNoSQLColumn(value);
+        return convertAnyToRowValue(formatted, !nested);
+      } else {
+        return {
+          value: RowValue.fromPartial({
+            stringValue: JSON.stringify(value),
+          }),
+          type: "TEXT",
+        };
+      }
+    }
+    default:
+      return {
+        value: RowValue.fromPartial({
+          stringValue: JSON.stringify(value),
+        }),
+        type: "TEXT",
+      };
+  }
+};
+
+export const flattenNoSQLResult = (resultSet: SQLResultSetV1) => {
+  for (const result of resultSet.results) {
+    const { columns, columnIndexMap } = getNoSQLColumns(result.rows);
+
+    const rows: QueryRow[] = [];
+    const columnTypeNames: string[] = Array.from({
+      length: columns.length,
+    }).map((_) => "TEXT");
+
+    for (const row of result.rows) {
+      if (row.values.length !== 1 || !row.values[0].stringValue) {
+        continue;
+      }
+      const data = JSON.parse(row.values[0].stringValue);
+      const values: RowValue[] = Array.from({ length: columns.length }).map(
+        (_) =>
+          RowValue.fromPartial({
+            nullValue: NullValue.NULL_VALUE,
+          })
+      );
+
+      for (const [key, value] of Object.entries(data)) {
+        const index = columnIndexMap.get(key) ?? 0;
+
+        const { value: formatted, type } = convertAnyToRowValue(value, true);
+        values[index] = formatted;
+        columnTypeNames[index] = type;
+      }
+
+      rows.push(
+        QueryRow.fromPartial({
+          values: values,
+        })
+      );
+    }
+
+    result.rows = rows;
+    result.columnNames = columns;
+    result.columnTypeNames = columnTypeNames;
+  }
+};
+
+const getNoSQLColumns = (rows: QueryRow[]) => {
+  const columnSet = new Set<string>();
+  const columnIndexMap = new Map<string, number>();
+
+  for (const row of rows) {
+    const parsedRows = getNoSQLRows(row);
+    if (!parsedRows) {
+      continue;
+    }
+    for (const item of parsedRows) {
+      columnSet.add(item.key);
+    }
+  }
+
+  const builtInColumns = new Map<string, number>([
+    ["_rid", 0],
+    ["_self", 1],
+    ["_etag", 2],
+    ["_attachments", 3],
+    ["_ts", 4],
+  ]);
+
+  const sortedColumns = orderBy(
+    [...columnSet],
+    [
+      (column) => (column === "id" || column === "-id" ? -1 : 1),
```

Comment:
column === "id" || column === "_id" ? I am not sure if it is a typo.

Reply:
You found a typo!